weave/artifact_wandb.py

```diff
@@ -647,6 +647,7 @@ def save(
         branch: typing.Optional[str] = None,
         *,
         _lite_run: typing.Optional["InMemoryLazyLiteRun"] = None,
+        _create_new_artifact: typing.Optional[bool] = True,
```

Comment:
sorry to double nit, but shouldn't this be `create_new_artifact_type`?

Reply:
I think it makes more sense to have a boolean that tells us what path we're in: whether we're creating a brand new artifact (publish_new, publish_as) or creating an artifact version inside of it (commit). The default case is a brand new artifact in which case we will upsert the project, create artifact types, do the run history keys query, etc.